### DISCUSSION TITLE
Can't pass vectors arguments such as `std::vector<std::string>` to commands in v2.0

### DISCUSSION THREAD

**Discussion by altairwei** [daniele77/cli#146] [Q&A] [answered]
```
/xxxxx/cli/include/cli/detail/fromstring.h:268:23: error: no match for ‘operator>>’ (operand types are ‘std::stringstream {aka std::__cxx11::basic_stringstream<char>}’ and ‘std::vector<std::__cxx11::basic_string<char>, std::allocator<std::__cxx11::basic_string<char> > >’)
         !(interpreter >> result) ||
          ~~~~~~~~~~~~~^~~~~~~~~~
```

**Comment by daniele77** [comment]
Can you please provide also the code that generates the compilation error?
Thank you.

**Comment by altairwei** [comment]
A simple modification to the example `simplelocalsession.cpp` can reproduce this problem:

```C++
    rootMenu -> Insert(
            "answer",
            [](std::ostream& out, int x, std::vector<std::string> strs){ out << "The answer is: " << x << "\n"; },
            "Print the answer to Life, the Universe and Everything " );
```

**Comment by daniele77** [comment, marked as answer]
Unfortunately mixing `std::vector<std::string>` with other parameter types is not currently supported.

As a workaround, you can use the first parameter of the vector as an int, by converting it from `string` in your code.

However, if you think this can be a useful feature, you can open an issue with a feature request (maybe you can also consider sponsoring the project using the heart icon on top of the page to speed up this development).
